playground/src/main/java/org/navgurukul/playground/ui/PlaygroundActivity.kt

```diff
@@ -96,6 +122,24 @@ class PlaygroundActivity : AppCompatActivity() {
             tvOutput.text = "" // clear output window
             viewModel.start(etCode.text.toString())
         }
+
+        etCode.addTextChangedListener( object: TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+                // cache text to shared Pref
+                sharedPreferences.edit().putString(KEY_PREF_CODE_BACKUP, etCode.text.toString()).apply()
```

Comment:
This is should be delegated via viewModel. We can create sharedPrefsProvider class and inject it via koin into viewModel to keep viewModel clean from Android Context class

Reply:
Done